assignments/callbacks.js

```diff
@@ -29,23 +29,78 @@ function getLength(arr, cb) {
   // getLength passes the length of the array into the callback.
 }
 
+function getLength(array, callback) {
+  callback(array.length);
+}
+getLength(items, (length) => {
+  alert(length);
+});
+// -------------------------------------------- //
+
 function last(arr, cb) {
   // last passes the last item of the array into the callback.
 }
 
+function last(arr, cb) {
+  return cb(arr[3]);
+}
+
+last(items, function(last) {
+  console.log(last)
+});
+// ------------------------------------------ //
+
+
 function sumNums(x, y, cb) {
   // sumNums adds two numbers (x, y) and passes the result to the callback.
 }
 
+function sumNums(x){
+  return function(y){
+    return x + y;
+  }
+  }
+  const solve = sumNums(x);
+  console.log(solve(y));
+  
+  // ------------------------------------------ //  
+
 function multiplyNums(x, y, cb) {
   // multiplyNums multiplies two numbers and passes the result to the callback.
 }
+ 
+function multiplyNums(x,y){
+return x * y;
+}
+
+function addNums(x,y){
+  return x + y;
+  }
+
+  function calculate(x,y,cb){
+    return  cb (x * y);
+    }
+
+const result = calculate(x,y,cb)
```

Comment:
Here, you are invoking the `calculate` function. The arguments you passed into the function are `x`, `y`, and `cb`. But these variables are not defined, leading to an error.

You'd either have to define in the surrounding scope, which is global in this case, what the values of these variables are, or to skip the variables altogether and pass the values directly into the function invocation:

```js
const result = calculate(2, 3, (product) => {
    console.log(`The product is ${product}`);
    return product;
});
```